subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompilerArgumentProvider.java

```diff
@@ -17,16 +17,13 @@
 package org.gradle.api.tasks.compile;
 
 import org.gradle.api.Incubating;
+import org.gradle.process.CommandLineArgumentProvider;
 
 /**
  * Provides command-line arguments to the compiler.
  *
  * @since 4.5
  */
 @Incubating
-public interface CompilerArgumentProvider {
-    /**
-     * The command-line arguments which will be provided to the compiler.
-     */
```

Comment:
Removing this Javadoc means that we currently say the compiler arguments will be provided to the "jvm", It's worth saving this more specific comment here I think.

Reply:
I am tempted to deprecate this interface directly and remove it in Gradle 4.7. WDYT?

Reply:
:+1: